api/jobs/handlers.py

```diff
@@ -234,6 +237,11 @@ def put(self, cid, rid):
             except APINotFoundException:
                 self.abort(400, 'Cannot find gear for alg {}, alg not valid'.format(updates['alg']))
 
+        if 'alg' in updates or 'config' in updates:
+            gear = get_gear_by_name(updates.get('alg', doc['alg']))
+            config_ = fill_gear_default_values(gear, updates.get('config', doc.get('config', {})))
+            validate_gear_config(gear, config_)
```

Comment:
I think `validate_gear_config` will throw an exception; maybe catch that and throw a @nagem -friendly exception for status codes & such?

Reply:
@kofalt The current (already friendly) exc thrown by it is `APIValidationException` (422). Wondering if that's appropriate or maybe `InputValidationException` is more fitting?